src/index.js

```diff
@@ -1,7 +1,163 @@
-console.log("CodeMoney42 - Sistema Bancário");
+import { AccountController } from './controller/AccountController.js';
+import readline from 'readline';
 
+class AccountApp {
+  constructor() {
+    this.accountController = new AccountController();
+    this.rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout
+    });
+    this.currentAccount = null;
+  }
+
+  start() {
+    this.showMainMenu();
+  }
+
+  showMainMenu() {
+    console.clear();
+    console.log('===== CODEMONEY BANK =====');
+    console.log('1. Criar Nova Conta');
+
+    if (this.accountController.getAccountCount() > 0) {
+      console.log('2. Selecionar Conta');
+    }
+
+    console.log('3. Sair');
+
+    this.rl.question('Escolha uma opção: ', (choice) => {
+      switch (choice) {
+        case '1':
+          this.createAccountPrompt();
+          break;
+        case '2':
+          if (this.accountController.getAccountCount() > 0) {
+            this.selectAccountPrompt();
+          } else {
+            console.log('Não há contas disponíveis.');
+            this.pause(this.showMainMenu);
+          }
+          break;
+        case '3':
+          this.exit();
+          break;
+        default:
+          console.log('Opção inválida!');
+          this.pause(this.showMainMenu);
+      }
+    });
+  }
+
+  createAccountPrompt() {
+    this.rl.question('Digite o nome do usuário: ', (name) => {
+      this.rl.question('Digite o saldo inicial (padrão 0): ', (initialBalance) => {
+        const balance = initialBalance ? parseFloat(initialBalance) : 0;
+
+        try {
+          const account = this.accountController.createAccount(name, balance)
+          this.currentAccount = account;
+          this.showAccountMenu();
+        } catch (error) {
+          console.error('Erro ao criar conta:', error.message);
+          this.pause(this.showMainMenu);
+        }
+      });

```

Comment:
error handling could be moved to the controller! it's one of the responsibilities it could take on.